chainstate/src/detail/tests/normal_tokens.rs

```diff
@@ -0,0 +1,837 @@
+};
+use chainstate_types::BlockIndex;
+use common::chain::block::BlockReward;
+use common::chain::config::TOKEN_MIN_ISSUANCE_FEE;
+use common::{
+    chain::{
+        block::{timestamp::BlockTimestamp, Block, ConsensusData},
+        signature::inputsig::InputWitness,
+        tokens::{token_id, OutputValue, TokenData, TokenId},
+        GenBlock, OutPointSourceId, OutputPurpose, Transaction, TxInput, TxOutput,
+    },
+    primitives::{time, Amount, Id, Idable},
+};
+use std::vec;
+
+enum ParentBlock {
+    BestBlock,
+    BlockId(Id<GenBlock>),
+}
+
+fn process_token(
+    test_framework: &mut TestFramework,
+    parent_block: ParentBlock,
+    values: Vec<OutputValue>,
+) -> Result<Option<BlockIndex>, BlockError> {
+    process_token_ex(test_framework, vec![], parent_block, values).map(|(_block_id, result)| result)
+}
+
+fn process_token_ex(
+    test_framework: &mut TestFramework,
+    additional_inputs: Vec<TxInput>,
+    parent_block: ParentBlock,
+    values: Vec<OutputValue>,
+) -> Result<(Block, Option<BlockIndex>), BlockError> {
+    let receiver = anyonecanspend_address();
+    let parent_block_id = match parent_block {
+        ParentBlock::BestBlock => test_framework.best_block_id(),
+        ParentBlock::BlockId(block_id) => block_id,
+    };
+    let test_block_info = TestBlockInfo::from_id(&test_framework.chainstate, parent_block_id);
+
+    // Create a token issue transaction and block
+    let mut inputs: Vec<TxInput> = test_block_info
+        .txns
+        .iter()
+        .flat_map(|(outpoint_source_id, outputs)| {
+            outputs
+                .iter()
+                .enumerate()
+                .map(|(output_index, _output)| {
+                    TxInput::new(
+                        outpoint_source_id.clone(),
+                        output_index.try_into().unwrap(),
+                        InputWitness::NoSignature(None),
+                    )
+                })
+                .collect::<Vec<TxInput>>()
+        })
+        .collect();
+
+    inputs.extend(additional_inputs);
+
+    let outputs: Vec<TxOutput> = values
+        .into_iter()
+        .map(|value| TxOutput::new(value, OutputPurpose::Transfer(receiver.clone())))
+        .collect();
+
+    let block = Block::new(
+        vec![Transaction::new(0, inputs, outputs, 0).unwrap()],
+        parent_block_id,
+        BlockTimestamp::from_duration_since_epoch(time::get()),
+        ConsensusData::None,
+        BlockReward::new(Vec::new()),
+    )
+    .unwrap();
+
+    // Process it
+    test_framework
+        .process_block(block.clone(), BlockSource::Local)
+        .map(|result| (block, result))
+}
+
+#[test]
+fn token_issue_test() {
+    common::concurrency::model(|| {
+        // Process token without errors
+
+        let mut test_framework = TestFramework::default();
+        // let mut chainstate = setup_chainstate();
+        let values = vec![OutputValue::Token(TokenData::TokenIssuanceV1 {
+            token_ticker: b"USDC".to_vec(),
```

Comment:
Change USDC to ABCD. I'm 99.99% sure we won't run in to issues here and sure it's a test but given it's probably a trademark (that I'm too lazy to check for) I'd rather err on the side of caution

Reply:
Just left everywhere - `SOME` ticker, I think this is enough